### PR TITLE
init: Validate minimum required cmake version

### PR DESCRIPTION
**Pull request by florianGla**

### CODE REVIEW

File: internal/cmd/init/init.go [hunk 1]
```diff
@@ -117,3 +137,39 @@ code coverage right from your IDE:
 		}
 	}
 }
+
+func validateMinimumRequiredCMakeVersion(projectDir string) (bool, error) {
+	versionPattern := regexp.MustCompile(
+		`(?is)cmake_minimum_required\([\n\t ]*version[\n\t ]*(?P<version>\d+(\.\d+){0,3})[\n\t ]*\)`,
```
Comment (jochil) [inline internal/cmd/init/init.go:143]: I did a little bit more research, as I am not very convinced of the regex "parsing"... maybe this command is an alternative
```
cmake --trace-source CMakeLists.txt --trace-format=json-v1 -B build
{"version":{"major":1,"minor":1}}
{"args":["VERSION","3.16"],"cmd":"cmake_minimum_required","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":1,"time":1660034973.0774741}
{"args":["cmake_example"],"cmd":"project","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":2,"time":1660034973.0775449}
-- The C compiler identification is GNU 12.1.1
-- The CXX compiler identification is GNU 12.1.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/sbin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/sbin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
{"args":[],"cmd":"enable_testing","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":4,"time":1660034973.3777411}
{"args":["cifuzz"],"cmd":"find_package","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":6,"time":1660034973.37781}
{"args":[],"cmd":"enable_fuzz_testing","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":7,"time":1660034973.3965089}
{"args":["src"],"cmd":"add_subdirectory","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":9,"time":1660034973.396575}
{"args":["exploreMe","explore_me.cpp"],"cmd":"add_library","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/src/CMakeLists.txt","frame":1,"line":1,"time":1660034973.3971341}
{"args":["exploreMe","PUBLIC","explore_me.h"],"cmd":"target_sources","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/src/CMakeLists.txt","frame":1,"line":2,"time":1660034973.3972681}
{"args":["${PROJECT_NAME}","main.cpp"],"cmd":"add_executable","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":11,"time":1660034973.397305}
{"args":["${PROJECT_NAME}","PRIVATE","exploreMe"],"cmd":"target_link_libraries","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":12,"time":1660034973.3974011}
{"args":["my_fuzz_test","my_fuzz_test.cpp"],"cmd":"add_fuzz_test","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":14,"time":1660034973.397429}
{"args":[],"cmd":"else","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":120,"time":1660034973.397552}
{"args":["CIFUZZ_SANITIZERS"],"cmd":"elseif","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":146,"time":1660034973.397598}
{"args":["my_fuzz_test","PRIVATE","exploreMe"],"cmd":"target_link_libraries","file":"/home/jochen/dev/workspace/ci/cifuzz/examples/cmake/CMakeLists.txt","frame":1,"line":15,"time":1660034973.397665}
-- Configuring done
-- Generating done
-- Build files have been written to: /home/jochen/dev/workspace/ci/cifuzz/examples/cmake/build
```

File: internal/cmd/init/init.go [hunk 2]
```diff
@@ -72,6 +76,22 @@ func setUpAndMentionBuildSystemIntegrations(cwd string) {
 		return
 	}
 	if cfg.BuildSystem == config.BuildSystemCMake {
+		versionValid, err := validateMinimumRequiredCMakeVersion(cwd)
+		if err != nil {
+			if !errors.Is(err, os.ErrNotExist) {
+				log.Debug(err)
+			}
+			return
+		}
```
Comment (jochil) [inline internal/cmd/init/init.go:85]: IMO it is not very intuitive that I have to check the error `validateMinimumRequiredCMakeVersion` for an `os.ErrNotExist`. I think this check should be done in the function itself... it is more like an implementation detail of the validation

**Comment by fmeum** [comment]
I agree with @jochil that checking the minimum version by parsing `CMakeLists.txt` ourselves is error-prone. 

I would suggest doing the check in our CMake package by parsing the [CMAKE_MINIMUM_REQUIRED_VERSION](https://cmake.org/cmake/help/latest/variable/CMAKE_MINIMUM_REQUIRED_VERSION.html) variable. We can pair on that if you want to get you started.

**Comment by jochil** [comment]
> I agree with @jochil that checking the minimum version by parsing `CMakeLists.txt` ourselves is error-prone.
> 
> I would suggest doing the check in our CMake package by parsing the [CMAKE_MINIMUM_REQUIRED_VERSION](https://cmake.org/cmake/help/latest/variable/CMAKE_MINIMUM_REQUIRED_VERSION.html) variable. We can pair on that if you want to get you started.

In this case the user will get notify the first time he runs `cmake` with our package included, or?
I think this is should be ok... and the package is definitely the best place for this kind of check

**Review by fmeum — APPROVED** [review]

**Pull request merged**